pkg/network/ndmz/ndmz.go

```diff
@@ -135,12 +136,37 @@ func Create(nodeID pkg.Identifier) error {
 			return errors.Wrapf(err, "ndmz: failed to enable enable_defrtr=1 in ndmz namespace")
 		}
 		// run DHCP to interface public in ndmz
-		received, err := dhcp.Probe(types.PublicIface, netlink.FAMILY_V4)
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+		probe := dhcp.NewPrope(ctx)
+
+		defer func() {
+			cancel()
```

Comment:
I noticed that this `cancel` is the only place where it's actually called. Hence there is not other trigger to stop this routing from finishing that that it actually just finish execution. 

I am not sure this is the desired scenario, but if that is the case, then you can just directly use context.Background() or no context at all!

Reply:
It is actually used line 158 to timeout and exist the loop there. But since context is just not use by `dhcp.Probe` I can just replace this ctx with a `time.After()` and be done with it.